### PR TITLE
Update `identity_entity` to use Sets for Policies

### PR DESCRIPTION
**Pull request by lawliet89**

### CODE REVIEW

**Comment by nazarewk** [comment]
+1 for this, looks like extremely good idea

File: vault/resource_identity_entity_test.go [hunk 1]
```diff
@@ -138,10 +140,18 @@ func testAccIdentityEntityCheckAttrs(entity string) resource.TestCheckFunc {
...
+
 					for i := 0; i < count; i++ {
-						stateData := instanceState.Attributes[stateAttr+"."+strconv.Itoa(i)]
-						if stateData != apiData[i] {
-							return fmt.Errorf("expected item %d of %s (%s in state) of %q to be %q, got %q", i, apiAttr, stateAttr, path, stateData, apiData[i])
+						found := false
+						for stateKey, stateValue := range instanceState.Attributes {
+							if strings.HasPrefix(stateKey, stateAttr) {
+								if apiData[i] == stateValue {
+									found = true
```
Comment (tyrannosaurus-becks) [inline vault/resource_identity_entity_test.go]: Should we break out of the loop here?

Comment (lawliet89) [reply vault/resource_identity_entity_test.go]: Good point.

Comment (lawliet89) [reply vault/resource_identity_entity_test.go]: That's done @tyrannosaurus-becks

Comment (tyrannosaurus-becks) [reply vault/resource_identity_entity_test.go]: :rainbow:

**Pull request merged**